firewood/src/merkle.rs

```diff
@@ -248,7 +232,7 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                             node_to_split,
                             |u| {
                                 match &mut u.inner {
-                                    NodeType::Leaf(u) => u.1 = Data(val),
+                                    NodeType::Leaf(u) => u.data = Data(val),
```

Comment:
nit: renaming `LeafNode(key, data)` to `LeafNode{key, data}` is great, but should be in a seperate diff.